### PR TITLE
use the hatch

### PR DESCRIPTION
**Pull request by bloodearnest**
- Add pydantic and itsdangerous dependencies
- Upload to release-hatch rather than directly to job-server

### CODE REVIEW

File: publisher/schema.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from datetime import datetime
```
Comment (sebbacon) [inline publisher/schema.py:1]: Is there something we should/can do to signpost that this schema lives in more than one place, until the time we factor it out into another library?

Comment (bloodearnest) [reply publisher/schema.py:1]: other than a comment?

Comment (sebbacon) [reply publisher/schema.py:1]: That's what I had in mind, but there might be better ideas!

Comment (bloodearnest) [reply publisher/schema.py:1]: https://github.com/opensafely-core/release-hatch/pull/18

File: publisher/upload.py [hunk 2]
```diff
@@ -1,51 +1,78 @@
...
+
+    release_id = response.headers["Release-Id"]
+    release_url = response.headers["Location"]
+
+    try:
+        for f in files:
+            release_file = ReleaseFile(name=str(f))
+            do_post(release_url, release_file.json(), auth_token)
+    except Forbidden:
+        # they can create releases, but not uploaded them
```
Comment (ghickman) [inline publisher/upload.py:47]: s/uploaded/upload/

**Review by ghickman — APPROVED** [review]
👍

**Pull request merged**